### DISCUSSION TITLE
C_GetOperationState Error when performing mTLS with s_server

### DISCUSSION THREAD

**Discussion by WolfDan** [openssl-projects/pkcs11-provider#299] [Q&A]
Hey! 

I'm trying to check an mtls connection with `s_server`, `s_client`, and the PKCS11 provider, but the `s_server` throws a `C_GetOperationState` error when the client is trying to do a post handshake new session; here's an image of the process and the error  

![image](https://github.com/latchset/pkcs11-provider/assets/5377526/589c4c51-3896-4c00-884a-22a83a775516)

Where the `s_server` uses an intermediate EC prime256v1 private key from PKCS11 and `s_client` uses an end entity private key from pkcs11 (the end entity was signed by the intermediate)

I've seen the `C_GetOperationState` error many times, but this is the first time this actually crashed something, so here's the debug log of `s_server` using the `-debug` flag 

![image](https://github.com/latchset/pkcs11-provider/assets/5377526/0bfa4c60-c40d-4634-a7b5-e1a35d08c62e)

Regarding the driver I'm using is tpm2-pkcs11 library 

I'm unsure what's the root of the issue at all, any idea what it might be? 

Thank you!

**Comment by simo5** [comment]
Images of text are not very useful ... can you please set PKC11_PROVIDER_DEBUG and get a debug file?
C_GetOperationState is generally optional, so I am surprised as well and need more info.
